### ISSUE TITLE
diffChangelog generates ADD CONSTRAINT changesets for existing unique constraints

### ISSUE THREAD

**Issue by cypressious** [liquibase/liquibase#2069]
## Environment
**Liquibase Version**: 4.4.3

**Liquibase Integration & Version**: Gradle

**Liquibase Extension(s) & Version**: Liquibase-Hibernate5 4.4.3

**Database Vendor & Version**: PostgreSQL 13

**Operating System Type & Version**: Windows 10, DB is running in Docker container

## Description
I have a Spring Boot application with Spring Data JPA connecting to a PostgreSQL 13 and I'm using Liquibase for migrations using the Gradle plugin.

When I run the `diffChangelog` Gradle task, the generated changelog contains statements like

```
-- changeset krakhman:1630671193066-3
ALTER TABLE public.surgery_free_friday ADD CONSTRAINT surgery_free_friday_date UNIQUE (date);
```

even though the `surgery_free_friday_date` is present in the database.

I've debugged the code and I've landed on the following line in `liquibase.snapshot.jvm.UniqueConstraintSnapshotGenerator#listColumns`:

![grafik](https://user-images.githubusercontent.com/1191816/132003160-71e15878-88d6-4c00-aab8-8e89da8f00e7.png)

As you can see on the screenshot, there is a mismatch in the cache keys, supposedly caused by the "null" which in turn is caused by this line

![grafik](https://user-images.githubusercontent.com/1191816/132004500-fd00863d-2b7d-4841-8fc4-e5d93f783852.png)

as `row.get("TABLE_NAME")` always seems to return `null`.

## Steps To Reproduce
It seems the bug is triggered, when `bulkQuery` is set to `true` in this line

![grafik](https://user-images.githubusercontent.com/1191816/132004849-42d2ccb6-c84b-4b2b-9397-1b3637d75378.png)

## Actual Behavior
`ADD CONSTRAINT` changesets are generated in the changelog.

## Expected/Desired Behavior
NO `ADD CONSTRAINT` changesets are generated in the changelog.

┆Issue is synchronized with this [Jira Story](https://datical.atlassian.net/browse/LB-2130) by [Unito](https://www.unito.io)

**Comment by molivasdat** [comment]
Hi @cypressious Thanks for writing up this issue and taking the time to debug through the code the problem  and then creating a PR to fix it.  We will add this to the list of issues to process.

**Comment by StevenMassaro** [comment]
I am unable to reproduce this bug on Liquibase 4.6.0. It looks to me like this issue may have been resolved with this change: https://github.com/liquibase/liquibase/commit/9bab5bd03dc10fbcb2449c0e2f1a1088f97aff5d

**Comment by cypressious** [comment]
I can confirm the bug is fixed in 4.6.1.
